cloudant-http/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -268,7 +255,9 @@ public HttpConnection execute() throws IOException {
         while (retry && numberOfRetries-- > 0) {
             connection = connectionFactory.openConnection(url);
 
-            connection.setRequestProperty("User-Agent", USER_AGENT);
+            // Set the default user-agent, consuming libraries are expected to set the UA via an
+            // interceptor, see UserAgentInterceptor
+            connection.setRequestProperty("User-Agent", DEFAULT_USER_AGENT);
```

Comment:
I think our libraries should always add the interceptor so I'm not sure it is worth adding this default header as it is an extra call on every single request (although I agree it could be nice in the case things went wrong with our library interceptors but we should have tests covering that).